### PR TITLE
Add memory leak check for vioscsi IO/block_resize/hotplug tests

### PR DESCRIPTION
**Pull request by peixiu**
Select 3 cases to add memory-leak check steps:
1) fio_windows
2) block_resize_unplug
3) block_hotplug_scsi_hba

ID: 2128921
Signed-off-by: Peixiu Hou <phou@redhat.com>

### CODE REVIEW

**Comment by peixiu** [comment]
Test results:
python ConfigTest.py --testcase=fio_windows.aio_native --guestname=Win2019 --platform=x86_64 --clone=no --driveformat=virtio_blk,virtio_scsi --machines=q35
(1/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.fio_windows.aio_native.q35: STARTED
 (1/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.fio_windows.aio_native.q35: PASS (520.60 s)
 (2/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.fio_windows.aio_native.q35: STARTED
 (2/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.fio_windows.aio_native.q35: PASS (484.36 s)

python ConfigTest.py --testcase=block_hotplug_scsi_hba --guestname=Win2019 --platform=x86_64 --clone=no --driveformat=virtio_scsi --machines=q35
 (1/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_hotplug_scsi_hba.with_hotplug_on.q35: STARTED
 (1/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_hotplug_scsi_hba.with_hotplug_on.q35: PASS (247.50 s)
 (2/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_hotplug_scsi_hba.with_hotplug_off.q35: STARTED
 (2/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_hotplug_scsi_hba.with_hotplug_off.q35: PASS (281.77 s)

python ConfigTest.py --testcase=block_resize_unplug --guestname=Win2019 --platform=x86_64 --clone=no --driveformat=virtio_scsi --machines=q35
 (1/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_qmp.reboot_vm.q35: STARTED
 (1/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_qmp.reboot_vm.q35: PASS (282.21 s)
 (2/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_shell.reboot_vm.q35: STARTED
 (2/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_shell.reboot_vm.q35: PASS (311.36 s)
 (3/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_qmp.reboot_vm.q35: STARTED
 (3/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_qmp.reboot_vm.q35: PASS (275.33 s)
 (4/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_shell.reboot_vm.q35: STARTED
 (4/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_shell.reboot_vm.q35: PASS (303.71 s)

python ConfigTest.py --testcase=block_resize_unplug --guestname=Win2019 --platform=x86_64 --clone=no --driveformat=virtio_blk --machines=q35
 (1/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_qmp.reboot_vm.q35: STARTED
 (1/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_qmp.reboot_vm.q35: PASS (312.83 s)
 (2/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_shell.reboot_vm.q35: STARTED
 (2/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_shell.reboot_vm.q35: PASS (343.48 s)
 (3/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_qmp.reboot_vm.q35: STARTED
 (3/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_qmp.reboot_vm.q35: PASS (308.09 s)
 (4/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_shell.reboot_vm.q35: STARTED
 (4/4) Host_RHEL.m9.u1.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_shell.reboot_vm.q35: PASS (353.77 s)

**Comment by peixiu** [comment]
This patch depends on https://github.com/autotest/tp-qemu/pull/3554

Hi @menli820 @qingwangrh @vivianQizhu , Could you help to review it? 

Thanks~
Peixiu

File: qemu/tests/block_resize_unplug.py [hunk 1]
```diff
@@ -103,4 +104,5 @@ def _check_img_size(size):
     if reboot:
         _change_vm_power()
         _check_vm_status()
+    win_driver_utils.memory_leak_check(vm, test, params)
```
Comment (menli820) [inline qemu/tests/block_resize_unplug.py:107]: This should restrict ‘windows’ as this case also fit linux.
if is_windows:
    win_driver_utils.memory_leak_check(vm, test, params)

File: qemu/tests/block_hotplug_scsi_hba.py [hunk 2]
```diff
@@ -73,4 +74,5 @@ def rescan_hba_controller(session):
             test.fail('Found a new disk with virtio-scsi-pci.hotplug=off '
                       'before rescan scsi hba controller.')
         rescan_hba_controller(session)
+    win_driver_utils.memory_leak_check(vm, test, params)
```
Comment (menli820) [inline qemu/tests/block_hotplug_scsi_hba.py:77]: The same here.

File: qemu/tests/block_resize_unplug.py (hunk 1 above)
Comment (peixiu) [reply qemu/tests/block_resize_unplug.py]: Thanks, updated~

File: qemu/tests/block_hotplug_scsi_hba.py (hunk 2 above)
Comment (peixiu) [reply qemu/tests/block_hotplug_scsi_hba.py]: @menli820 Thanks, updated~

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=fio_windows.aio_native,block_hotplug_scsi_hba,block_resize_unplug --guestname=Win2019 --platform=x86_64  --driveformat=virtio_scsi --machines=q35  --firmware=default_bios

failed case please see http://fileshare.englab.nay.redhat.com/pub/section2/images_backup/qbugs/2128921/2022-12-22/latest/results.html

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=fio_windows.aio_native,block_hotplug_scsi_hba,block_resize_unplug --guestname=Win2019 --platform=x86_64 --driveformat=virtio_scsi --machines=q35 --firmware=default_bios --clone=no
(1/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.fio_windows.aio_native.q35: STARTED
 (1/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.fio_windows.aio_native.q35: PASS (187.75 s)
 (2/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_qmp.reboot_vm.q35: STARTED
 (2/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_qmp.reboot_vm.q35: PASS (158.88 s)
 (3/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_shell.reboot_vm.q35: STARTED
 (3/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_shell.reboot_vm.q35:PASS (178.92 s)
 (4/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_qmp.reboot_vm.q35: STARTED
 (4/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_qmp.reboot_vm.q35: PASS (158.99 s)
 (5/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_shell.reboot_vm.q35: STARTED
 (5/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_shell.reboot_vm.q35: PASS (177.22 s)
 (6/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_hotplug_scsi_hba.with_hotplug_on.q35: STARTED
 (6/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_hotplug_scsi_hba.with_hotplug_on.q35: PASS (165.08 s)
 (7/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_hotplug_scsi_hba.with_hotplug_off.q35: STARTED
 (7/7) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_hotplug_scsi_hba.with_hotplug_off.q35: PASS (199.01 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0


LGTM

**Comment by menli820** [comment]
LGTM

 (1/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.fio_windows.aio_native.q35: STARTED
 (1/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.fio_windows.aio_native.q35: PASS (175.40 s)
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.fio_windows.aio_threads.q35: STARTED
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.fio_windows.aio_threads.q35: PASS (171.95 s)

 (1/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_qmp.reboot_vm.q35: STARTED
 (1/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_qmp.reboot_vm.q35: PASS (167.03 s)
 (2/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_shell.reboot_vm.q35: STARTED
 (2/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_shell.reboot_vm.q35: PASS (201.19 s)
 (3/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_qmp.reboot_vm.q35: STARTED
 (3/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_qmp.reboot_vm.q35: PASS (162.46 s)
 (4/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_shell.reboot_vm.q35: STARTED
 (4/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_shell.reboot_vm.q35: PASS (200.52 s

**Comment by menli820** [comment]
LGTM

 (1/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.fio_windows.aio_native.q35: STARTED
 (1/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.fio_windows.aio_native.q35: PASS (175.40 s)
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.fio_windows.aio_threads.q35: STARTED
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.fio_windows.aio_threads.q35: PASS (171.95 s)

 (1/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_qmp.reboot_vm.q35: STARTED
 (1/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_qmp.reboot_vm.q35: PASS (167.03 s)
 (2/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_shell.reboot_vm.q35: STARTED
 (2/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.enlarge_disk.send_shell.reboot_vm.q35: PASS (201.19 s)
 (3/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_qmp.reboot_vm.q35: STARTED
 (3/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_qmp.reboot_vm.q35: PASS (162.46 s)
 (4/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_shell.reboot_vm.q35: STARTED
 (4/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.block_resize_unplug.shrink_disk.send_shell.reboot_vm.q35: PASS (200.52 s

**Review by vivianQizhu — APPROVED** [review]

**Pull request merged**